app/controllers/application_controller.rb

```diff
@@ -17,4 +19,11 @@ def configure_permitted_parameters
   def set_user
     UserInfo.current_user = authenticate_user!
   end
+
+  def check_authorization
+    authorize "#{request.params[:controller].titlecase.gsub(/\s+/, '')}Controller".constantize.new
+  rescue Pundit::NotAuthorizedError
+    flash[:warn] = t('shared.errors.authorisation')
+    redirect_to root_path
+  end
```

Comment:
Although it is great to reduce duplication for this method, previously access was restricted to defined actions per controller whereas this seems to check all controller actions... for example in `msa_components_controller` we had: `before_action :check_authorisation, only: %i(new create)`... this now means that a `user_manager` can't view components read only. If we add more detail into the policy classes for each action then all should be ok... I will try and do it tomorrow morning!

Reply:
Explained in Person